### ISSUE TITLE
C#7 change in behavior: can no longer mix comparison operator with bitwise operator

### ISSUE THREAD

**Issue by chucker** [dotnet/roslyn#17296]
**Version Used**: 
This issue occurs in VS Enterprise 2017 RC 15.0.0-RC.4+26206.0 (with presumably C# 7). It does not occur in 14.0.25341.01 Update 3 (C# 6).

**Steps to Reproduce**:

1. Use the following code:

```csharp
public static byte[] StringToByteArray(string hex)
{
    for (int i = 0; i < hex.Length >> 1; ++i)
    {
    }

    return new byte[] { };
}
```

2. Paste it into a Visual Studio 2015 C# project.
3. Also paste it into a Visual Studio 2017 RC project.

**Expected Behavior**:

Both should behave identically, or at least explain the difference.

**Actual Behavior**:

Visual Studio 2015 formats and compiles the code.

Visual Studio 2017 RC, however, formats it incorrectly:

```csharp
public static byte[] StringToByteArray(string hex)
{
    for (int i = 0; i<hex.Length> > 1; ++i)
    {
    }

    return new byte[] { };
}
```

(Note the space between `> >`, thus breaking the operator.)

And also fails to compile it. It apparently assumes the above syntax to be generics, thus failing with the following error messages:

CS0118	'hex' is a variable but is used like a type
CS0307	The variable 'i' cannot be used with type arguments

I actually appreciate that this needlessly ambiguous and hard-to-read code is no longer supported, but the error messages lead me to suspect this wasn't an intended change, but is rather a side-effect of a change in parsing.

**Comment by miloush** [comment]
Note that if you use constant number instead of name it works, that itself isn't good. Introducing local constant fix for example would break it. Except it is already broken - if you try to put a constant number there and introduce local const for it, the renaming refactoring marks it as a conflict.

**Comment by 0xd4d** [comment]
Perhaps this was fixed by #16834 ?

**Comment by KeesCBakker** [comment]
Possible work-a-round: `for (int i = 0; i < (hex.Length) >> 1; ++i)`
